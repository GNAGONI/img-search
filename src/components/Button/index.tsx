import Button from '@mui/material/Button';
import { styled } from '@mui/material/styles';

interface Props {
  title: string;
  onClick?: () => void;
}

const CustomStyledButton = styled(Button)({
  width: '100px',
  height: '56px',
  marginLeft: '4px',
});

const CustomButton = (props: Props) => {
  return (
    <CustomStyledButton
      type="submit"
      variant="contained"
      color="primary"
      onClick={props.onClick}
    >
      <p>{props.title}</p>
    </CustomStyledButton>
  );
};

export default CustomButton;
