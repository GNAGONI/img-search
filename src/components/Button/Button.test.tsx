import renderer from 'react-test-renderer';
import Button from './index';

it('renders correctly Button', () => {
  const mockCallback = jest.fn();
  const tree = renderer
    .create(<Button title="Test" onClick={mockCallback} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});