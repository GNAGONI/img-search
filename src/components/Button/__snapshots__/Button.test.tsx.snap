// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Button 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-q1b40h-MuiButtonBase-root-MuiButton-root"
  disabled={false}
  onBlur={[Function]}
  onClick={[MockFunction]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="submit"
>
  <p>
    Test
  </p>
</button>
`;
